build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildTasks.ts

```diff
@@ -351,6 +357,14 @@ function hasTaskDependency(
 	const rootConfig = loadFluidBuildConfig(root);
 	const globalTaskDefinitions = normalizeGlobalTaskDefinitions(rootConfig?.tasks);
 	const taskDefinitions = getTaskDefinitions(json, globalTaskDefinitions, false);
+	// Searched deps that are package specific
+	const packageSpecificSearchDeps = searchDeps.filter((d) => d.includes("#"));
```

Comment:
Hmm. I think that is less accurate.
1. The package specifier could point to this same package. I don't think they should, but it is possible.
2. Other dependencies might also be from other packages. ^ deps are also from other packages. So, this doesn't capture all of the other package dependencies.
This thing is exactly what it claims. I can add e.g. text though :)

Reply:
Fair points :). I don't think we use # for same-package deps in practice even though it seems technically possible (although I wouldn't be super confident the implementation actually works in that case), since we can just use the plain script name. Maybe `Searched task deps that are package specific (we normally use # to indicate dependencies on specific tasks of other packages which are dependencies of the one in question)` or something like that?